core/citrus-api/src/main/java/org/citrusframework/TestResult.java

```diff
@@ -30,216 +30,236 @@
+    /**
+     * Execution duration
+     */
+    private final Duration duration;
+
+    /**
+     * Failure stack trace
+     */
     private String failureStack;
 
-    /** Failure type information */
+    /**
+     * Failure type information
+     */
     private String failureType;
 
     /**
      * Create new test result for successful execution.
+     *
      * @param name
      * @param className
      * @return
      */
-    public static TestResult success(String name, String className) {
-        return new TestResult(name, className, RESULT.SUCCESS);
+    public static TestResult success(String name, String className, Duration duration) {
+        return new TestResult.Builder(RESULT.SUCCESS, name, className)
+                .setDuration(duration)
+                .build();
     }
 
     /**
      * Create new test result with parameters for successful execution.
+     *
      * @param name
      * @param className
      * @param parameters
      * @return
      */
-    public static TestResult success(String name, String className, Map<String, Object> parameters) {
-        return new TestResult(name, className, RESULT.SUCCESS, parameters);
+    public static TestResult success(String name, String className, Duration duration, Map<String, Object> parameters) {
+        return new TestResult.Builder(RESULT.SUCCESS, name, className)
+                .setDuration(duration)
+                .setParameters(parameters)
+                .build();
     }
 
     /**
      * Create new test result for skipped test.
+     *
      * @param name
      * @param className
      * @return
      */
     public static TestResult skipped(String name, String className) {
-        return new TestResult(name, className, RESULT.SKIP);
+        return new TestResult.Builder(RESULT.SKIP, name, className)
+                .build();
     }
 
     /**
      * Create new test result with parameters for skipped test.
+     *
      * @param name
      * @param className
      * @param parameters
      * @return
      */
     public static TestResult skipped(String name, String className, Map<String, Object> parameters) {
-        return new TestResult(name, className, RESULT.SKIP, parameters);
+        return new TestResult.Builder(RESULT.SKIP, name, className)
+                .setParameters(parameters)
+                .build();
     }
 
     /**
      * Create new test result for failed execution.
+     *
      * @param name
      * @param className
      * @param cause
      * @return
      */
-    public static TestResult failed(String name, String className, Throwable cause) {
-        return new TestResult(name, className, RESULT.FAILURE, cause);
+    public static TestResult failed(String name, String className, Throwable cause, Duration duration) {
+        return new Builder(RESULT.FAILURE, name, className)
+                .setCause(cause)
+                .setErrorMessage(Optional.ofNullable(cause).map(Throwable::getMessage).orElse(""))
```

Comment:
I don't like static imports on`Optional`'s, because they're so unreadable. with a little magic (refactoring the method) I managed to make it better 👍🏼